plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -15,14 +15,15 @@ import (
 
 // SQLServer struct
 type SQLServer struct {
-	Servers       []string `toml:"servers"`
-	QueryVersion  int      `toml:"query_version"`
-	AzureDB       bool     `toml:"azuredb"`
-	DatabaseType  string   `toml:"database_type"`
-	IncludeQuery  []string `toml:"include_query"`
-	ExcludeQuery  []string `toml:"exclude_query"`
-	queries       MapQuery
-	isInitialized bool
+	Servers             []string `toml:"servers"`
+	QueryVersion        int      `toml:"query_version"`
+	AzureDB             bool     `toml:"azuredb"`
+	DatabaseType        string   `toml:"database_type"`
+	IncludeQuery        []string `toml:"include_query"`
+	ExcludeQuery        []string `toml:"exclude_query"`
+	IgnoreDefaultServer bool     `toml:"ignore_default_server"`
```

Comment:
I don't understand why we need an "ignore default server" flag.

Reply:
Thank you @ssoroka for taking a look at this. The "ignore default server" flag was added for the case when the server list is empty and no server should be monitored. Without the flag, an empty server list would cause the localhost to be monitored, which would throw errors when using the agent for remote monitoring. Also, the localhost mechanism is valid only for SQL Server database type, but not for Azure SQL Database or Azure SQL Managed Instance. By adding the flag, the previous experience would work as is, and whoever does not want to monitor localhost can use this flag.

Reply:
I didn't spot that. tbh, I don't get why someone should leave the list empty even for localhost, I know the plugin will look for the default instance (port 1433) on localhost but I've never used that anyway as I want all to be explicitly set.

I'm not against the change, I just find it odd, if you don't want to use the plugin you can simply comment it out, or if you use multiple .conf file and the `--config-directory` you can simply change the file extension to something different than .conf and it will be ignored (I personally use this pattern `input_sqlserver.conf.ignore`.

In any case, this looks out of scope for this PR, as it should just add the new queries, can you propose this in a new  PR?

Reply:
This isn't how we normally do defaults anyway. the default should go in the init() function, and then you don't need conditional checks like 
```

	if len(s.Servers) == 0 {
		s.Servers = append(s.Servers, defaultServer)
	}
```
that assume empty lists are unset. Let's remove this boolean and move the default to the init() function, and then explicitly empty lists will work properly.

Reply:
Thank you @ssoroka and @Trovalo for the feedback. Let me pull this change out and create a separate PR for it, so that HA/DR queries related changes can go ahead and get merged.

Reply:
@ssoroka and @Trovalo - I've reverted the default flag related changes and will create a separate PR for it. This PR now contains only changes for HA/DR queries. Please let me know if you have further feedback on this. If not, it'd be great to get this merged. Thank you.

Reply:
@ssoroka - Could you please help merge this PR if there are no concerns?